lib/ts/Scrubber.cc

```diff
@@ -0,0 +1,201 @@
+/** @file
+
+  Implementation for Scrubber.h
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "Scrubber.h"
+
+Scrubber::Scrubber(const char *config)
+{
+  bool state_expecting_regex = true;
+  const ts::StringView delimiters("->;,", ts::StringView::literal);
+  ts::StringView regex;
+  ts::StringView replacement;
+  ts::StringView text(config);
```

Comment:
This is problematic, because the `StringView` will be pointing in to external memory. This needs to point to the local memory in `this->config`.